src/main/scala/intellij/haskell/notification/HaskellSDKNotificationProvider.scala

```diff
@@ -0,0 +1,73 @@
+package intellij.haskell.notification
+
+import com.intellij.ProjectTopics
+import com.intellij.openapi.fileEditor.FileEditor
+import com.intellij.openapi.project.{Project, ProjectBundle}
+import com.intellij.openapi.roots.ui.configuration.ProjectSettingsService
+import com.intellij.openapi.roots.{ModuleRootAdapter, ModuleRootEvent}
+import com.intellij.openapi.util.Key
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.ui.{EditorNotificationPanel, EditorNotifications}
+import intellij.haskell.external.component.StackProjectStartupManager
+import intellij.haskell.sdk.HaskellSdkType
+import intellij.haskell.util.HaskellFileUtil
+
+class HaskellSDKNotificationProvider(val myProject: Project, val notifications: EditorNotifications) extends EditorNotifications.Provider[EditorNotificationPanel] {
+  private val KEY: Key[EditorNotificationPanel] = Key.create("Setup Haskell Stack SDK")
+  private var currentHaskellSDKNameOption = HaskellSdkType.getCurrentHaskellSDKName(myProject)
+
+  myProject.getMessageBus.connect(myProject).subscribe(ProjectTopics.PROJECT_ROOTS, new ModuleRootAdapter() {
+    override def rootsChanged(event: ModuleRootEvent) {
+      notifications.updateAllNotifications()
+    }
+  })
+
+  override def getKey: Key[EditorNotificationPanel] = KEY
+
+  override def createNotificationPanel(file: VirtualFile, fileEditor: FileEditor): EditorNotificationPanel = {
+    if (HaskellFileUtil.isFileInProject(myProject, file)) {
```

Comment:
I do not understand this logic. Why do you check for file and next line for project. Is only project not enough?